docs/develop/secrets.md

```diff
@@ -1,9 +1,123 @@
 # Secret management
 
-{% hint style="warning" %}
+Some models need access to APIs, databases, AWS resources, or other secured information. With Truss, you can securely reference access keys, API tokens, passwords, secrets, and more. Keep this sensitive information out of your Git repository and use the same best practices for ML models as production websites.
+
+### Accessing secrets in models
+
+This example is based on the [GFP-GAN example Truss](https://github.com/basetenlabs/truss/tree/main/examples/gfpgan) in the main repository. That model needs access to an S3 bucket on AWS, requiring an access key id, access key secret, AWS region, and bucket name. These values are handled as secrets.
+
+{% hint style="danger" %}
+Never commit secret values to source control.
+{% endhint %}
+
+Start by defining a list of secrets in the `config.yaml` file. YAML is a key-value store, but you don't want to **ever** actually store the secret values in this file, even when running the Truss locally. Instead, set default values for the secrets, like `null`, and can leave comments about what kind of value to expect. Default values can also be fallback values that show format or type, like an empty string, the number 0, or an example UUID.
+
+```yaml
+secrets:
+  gfpgan_aws_access_key_id: null
+  gfpgan_aws_secret_access_key: null
+  gfpgan_aws_region: null # e.g. us-east-1
+  gfpgan_aws_bucket: null
+```
+
+Then, you can access the secrets in the `model/model.py` file by referencing them as kwargs in the init function.
+
+```python
+def __init__(self, **kwargs) -> None:
```

Comment:
A more readable way is to refer to secrets directly as a construction param:
```
def __init__(self, secrets) -> None:
```

btw, same goes for config and data_dir params.

```
def __init__(self, config) -> None:
```

```
def __init__(self, config, secrets) -> None:
```

```
def __init__(self, config, secrets, data_dir) -> None:
```

are all valid. 
We look for the params defined in the constructor and bind them accordingly by name.

Reply:
Agree this is more readable, but I am going to keep it exactly the same as the example code to avoid confusion if someone references the examples in the repository (which I link to)